### PR TITLE
Traverse.

### PR DESCRIPTION
**Pull request by CPWstatic**
#### What type of PR is this?
- [x] enhancement

#### What does this PR do?
Added an executor called `Traverse`

#### Which issue(s)/PR(s) this PR relates to?
Part work of implementing the openCypher.
  

#### Checklist：
- [x] Performance impacted:  We hope there are some improvement.

#### Release notes：
N/A                                                         `

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Generally LGTM, Only want to know what's difference between `GetVertices` and `AppendVertices` ?

File: src/graph/executor/query/GetVerticesExecutor.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ folly::Future<Status> GetVerticesExecutor::execute() { return getVertices(); }
 folly::Future<Status> GetVerticesExecutor::getVertices() {
   SCOPED_TIMER(&execTime_);
 
-  auto *gv = asNode<GetVertices>(node());
+  auto *gv = asNode<AppendVertices>(node());
```
Comment (Shylock-Hg) [inline src/graph/executor/query/GetVerticesExecutor.cpp:24]: Mistake?

**Comment by CPWstatic** [comment]
> Generally LGTM, Only want to know what's difference between `GetVertices` and `AppendVertices` ?

`AppendVertices` append vertices to a path.

File: src/graph/executor/query/GetVerticesExecutor.cpp (hunk 1 above)
Comment (CPWstatic) [reply src/graph/executor/query/GetVerticesExecutor.cpp]: Yes, thanks.

**Comment by yixinglu** [comment]
Please solve the conflicts!

File: src/common/datatypes/Edge.cpp [hunk 2]
```diff
@@ -131,6 +131,16 @@ void Edge::clear() {
   props.clear();
 }
 
+bool Edge::keyEqual(const Edge& rhs) const {
+  if (type != rhs.type && type != -rhs.type) {
+    return false;
+  }
+  if (type == rhs.type) {
+    return src == rhs.src && dst == rhs.dst && ranking == rhs.ranking;
```
Comment (Shylock-Hg) [inline src/common/datatypes/Edge.cpp:139]: EdgeType?

File: src/graph/validator/test/QueryValidatorTest.cpp [hunk 3]
```diff
@@ -1238,53 +1224,25 @@ TEST_F(QueryValidatorTest, TestMatch) {
...
-                                            PK::kPassThrough,
-                                            PK::kGetNeighbors,
-                                            PK::kStart,
-                                            PK::kDedup,
-                                            PK::kProject,
-                                            PK::kStart};
-    EXPECT_TRUE(checkResult(query, expected));
-  }
-  {
-    std::string query = "MATCH p = (n)-[]-(m:person{name:\"LeBron James\"}) RETURN p";
```
Comment (Shylock-Hg) [inline src/graph/validator/test/QueryValidatorTest.cpp:1272]: Why delete this case?

**Review by Shylock-Hg — APPROVED** [review]
LGTM

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**